htmresearch/frameworks/clustering/sdr_clustering.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import numpy as np
+
+from htmresearch.frameworks.clustering.distances import clusterDist
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+
+
+class Point(object):
+  def __init__(self, pointId, value, label):
+    self._id = pointId
+    self._value = value
+    self._label = label
+
+
+  def getId(self):
+    return self._id
+
+
+  def setLabel(self, label):
+    self._label = label
+
+
+  def getLabel(self):
+    return self._label
+
+
+  def setValue(self, value):
+    self._value = value
+
+
+  def getValue(self):
+    return self._value
+
+
+
+class Cluster(object):
+  def __init__(self, id, points=None, label=None):
+    self._id = id
+    self._label = label
+    if points:
+      self._points = points
+    else:
+      self._points = []
+    _LOGGER.debug("New cluster %s created!" % self._id)
+
+
+  def add(self, point):
+    self._points.append(point)
+
+
+  def setLabel(self, name):
+    self._label = name
+
+
+  def getLabel(self):
+    return self._label
+
+
+  def getId(self):
+    return self._id
+
+
+  def setPoints(self, points):
+    self._points = points
+
+
+  def getPoints(self):
+    return self._points
+
+
+  def size(self):
+    return len(self._points)
+
+
+  def merge(self, cluster):
+    for point in cluster.getPoints():
+      self.add(point)
```

Comment:
I'd rather implement what we talked about instead: add new point _only_ if you don't find any point in the cluster where `d(newPoint, point) < someThreshold`